dev/com.ibm.ws.security.oidc.social_fat.backchannelLogout/fat/src/com/ibm/ws/security/oidc_social/backchannelLogout/fat/LogoutTokenCreationTests.java

```diff
@@ -0,0 +1,561 @@
+            }
+        } else {
+            // need to have validate against what was in the id_token
+            Object oneIdTokenValue = idTokenDataClaims.get(oneKey);
+            if (oneIdTokenValue == null) {
+                fail("Key: " + oneKey + " was missing in the id_token (nothing to validate the logout_token value against");
+            }
+            if (!oneIdTokenValue.toString().equals(oneLogoutTokenValue.toString())) {
+                fail("Was epecting Key: " + oneKey + " with value: " + oneIdTokenValue.toString() + ", but found: " + oneLogoutTokenValue.toString());
+            }
+        }
+
+        // don't have to check for null - if it's null, we don't need to check it and we've already checked for both claims being null
+        if (otherLogoutTokenValue != null) {
+            // need to have validate against what was in the id_token
+            Object otherIdTokenValue = idTokenDataClaims.get(otherKey);
+            if (otherIdTokenValue == null) {
+                fail("Key: " + otherKey + " was missing in the id_token (nothing to validate the logout_token value against");
+            }
+
+            if (!otherIdTokenValue.toString().equals(otherLogoutTokenValue.toString())) {
+                fail("Was epecting Key: " + otherKey + " with value: " + otherIdTokenValue.toString() + ", but found: " + otherLogoutTokenValue.toString());
+            }
+
+        }
+
+    }
+
+    protected void mustNotHave(String key, Map<String, Object> logoutClaims) throws Exception {
+
+        Object logoutTokenValue = logoutClaims.get(key);
+        if (logoutTokenValue != null) {
+            fail("Key: " + key + " was included in the logout_token and should NOT have been");
+        }
+    }
+
+    protected void mustHaveEvents(Map<String, Object> logoutClaims) throws Exception {
+
+        Object rawLogoutTokenValue = logoutClaims.get(Constants.PAYLOAD_EVENTS);
+        if (rawLogoutTokenValue == null) {
+            fail("Key: events is missing in the logout_token");
+        }
+
+        if (rawLogoutTokenValue instanceof JsonObject) {
+
+            Log.info(thisClass, "ugh", "in jsonObject");
+            Log.info(thisClass, "mustHaveEvents", "JsonObject: " + rawLogoutTokenValue);
+            JSONObject event = JSONObject.parse(rawLogoutTokenValue.toString());
+            if (event == null) {
+                fail("Key: events is Null in the logout_token");
+            }
+            if (!event.containsKey(logoutEventKey)) {
+                fail("Key: events in the logout_token does not contain a Json object containing the key: " + logoutEventKey);
+            }
+            Object value = event.get(logoutEventKey);
+            if (value == null) {
+
+            } else {
+                if (!value.equals(new JSONObject())) {
+                    fail("value for the: " + logoutEventKey + " within events is NOT an empty Json object - the value is: " + value);
```

Comment:
@ayoho - Yeah - I can just log a message if we find it - will our OP ever not return an empty JSON object?
I have to balance spec vs what we'll do - these tests are testing our OP's behavior and we never plan on putting anything in there, I should validate that (since it is what the spec allows)

Reply:
I'll leave this at just issuing a message, bug pass the check.